### ISSUE TITLE
Issue with new update

### ISSUE THREAD

**Issue by Cerobug** [MatthieuBonne/CQAutomater#26]
Getting this error and the automater is crashing after ~10 seconds of being opened.

`9/16/2019 7:53:03 PM
System.AggregateException: One or more errors occurred. ---> System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.WebException: The request was aborted: Could not create SSL/TLS secure channel.
   at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   at System.Net.Http.HttpClientHandler.GetResponseCallback(IAsyncResult ar)
   --- End of inner exception stack trace ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at LinqToTwitter.Net.GetMessageHandler.<SendAsync>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at LinqToTwitter.TwitterExecute.<QueryTwitterAsync>d__48`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at LinqToTwitter.TwitterContext.<ExecuteAsync>d__136`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at LinqToTwitter.TwitterQueryProvider.<ExecuteAsync>d__8`1.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at LinqToTwitter.TwitterQueryable`1.GetEnumerator()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at CQFollowerAutoclaimer.AutoEvent.GetTwitterFeeds()
   at CQFollowerAutoclaimer.AutoEvent.<CouponTimer_Elapsed>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
---> (Inner Exception #0) System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.WebException: The request was aborted: Could not create SSL/TLS secure channel.
   at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   at System.Net.Http.HttpClientHandler.GetResponseCallback(IAsyncResult ar)
   --- End of inner exception stack trace ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at LinqToTwitter.Net.GetMessageHandler.<SendAsync>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at LinqToTwitter.TwitterExecute.<QueryTwitterAsync>d__48`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at LinqToTwitter.TwitterContext.<ExecuteAsync>d__136`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at LinqToTwitter.TwitterQueryProvider.<ExecuteAsync>d__8`1.MoveNext()<---

`

**Comment by MatthieuBonne** [comment]
For some reason your CQA can't connect to Twitter. Next commit will include a failsafe to prevent crashes.

**Comment by Vitae666** [comment]
Here's my error from the current commit. Ironically it happened almost 5 minutes after I walked out the door.

```
9/17/2019 2:14:44 PM
System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.Select[TSource,TResult](IEnumerable`1 source, Func`2 selector)
   at CQFollowerAutoclaimer.AutoEvent.<EventTimer_Elapsed>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()

```

**Comment by MatthieuBonne** [comment]
Did it happen several times ?

**Comment by Vitae666** [comment]
2 time after that.
But then because of MC and Adventure both running constantly, I shut Events off and it was fine after that.
And nope, no crash today since I re-enabled CQA. At least none yet.

**Comment by djozz911** [comment]
I have some issues here too :

1./ The PG event didn't start until I picked a card. I haven't unchecked the auto event in CQA so it started to pick cards until all attempts were used.

25/09/2019 08:47:46	 Picking PG cell 1
25/09/2019 08:48:17	 Picking PG cell 1
25/09/2019 08:48:49	 Picking PG cell 1
25/09/2019 08:51:31	 Picking PG cell 1
25/09/2019 08:52:03	 Picking PG cell 1
25/09/2019 08:52:35	 Picking PG cell 1
25/09/2019 08:53:07	 Picking PG cell 1
25/09/2019 08:53:39	 Picking PG cell 1
25/09/2019 08:54:11	 Picking PG cell 1
25/09/2019 08:54:43	 Picking PG cell 1
25/09/2019 08:55:14	 Picking PG cell 1
25/09/2019 08:55:47	 Picking PG cell 1
25/09/2019 08:56:18	 Picking PG cell 1

2./ Today, CQA detected the Adventure event as active but didn't started it. No entries in action or error log.

3./ I have these entries in the error log 

25/09/2019 09:23:01
25/09/2019 09:23:04
	Error Cloud Script Error: status PlayFab.ClientModels.ExecuteCloudScriptResult
25/09/2019 13:50:47
25/09/2019 13:50:50
	Error Cloud Script Error: status PlayFab.ClientModels.ExecuteCloudScriptResult
25/09/2019 15:47:26
	Error JsonParseError Invalid JSON string
25/09/2019 19:09:56
25/09/2019 19:09:58
	Error Cloud Script Error: status PlayFab.ClientModels.ExecuteCloudScriptResult

**Comment by MatthieuBonne** [comment]
1) and 2) look similar to me, I've had this bug but not every week... Trying to find why.
3) will be fixed in next version.

**Comment by Vitae666** [comment]
Yeah, that's all detailed in my MC gone wild post :)
Also, Adventure doesn't do anything. Yet. At some time Matt will update it and add stuff. Unless I'm mistaken, and it's not doing anything for me either :)

**Comment by djozz911** [comment]
PG event didn't start today. I'm using v4.5
No entry log.

**Comment by Cerobug** [comment]
It's crashing again after updating today. Here's the error I am seeing before it crashes.

10/4/2019 11:48:25 PM
	Debug in PGCards, PGCards attempts remaining : no
10/4/2019 11:49:27 PM
	Debug in PGCards, PGCards attempts remaining : no
10/4/2019 11:54:37 PM
	Debug in PGCards, PGCards attempts remaining : no

10/4/2019 11:57:56 PM
System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.Select[TSource,TResult](IEnumerable`1 source, Func`2 selector)
   at CQFollowerAutoclaimer.AutoEvent.<EventTimer_Elapsed>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_1(Object state)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()

**Comment by djozz911** [comment]
2 things :

- CQA didn't started Adventure event but detected it, I've started an AS quest manually. Upon completion, CQA started a PG quest and an AS quest after that. Is it possible to only start AS Quest ? Or eventually make a string like 'Quest 2, Quest 2, Quest 1' ?

edit : I looked at the code and see that "if AS<100 then start PG". So if I start the adventure with less than 200 AS (180 in my case), CQA doesn't take into account the win I get (100+rng) and considers I only have 80 AS and start a PG. Is a xx seconds timer or a forced currency update could resolve this issue ?

Action.log
14/10/2019 17:19:14	 Sending Adventure (1, 100%)
15/10/2019 01:09:44	 Sending Adventure (2, 100%)

- Running for 7d23h43m... CQA uses a lot of memory.

![image](https://user-images.githubusercontent.com/47937938/66803972-88d30200-ef21-11e9-8689-c5f16e3ee1e2.png)

**Comment by MatthieuBonne** [comment]
Adventure : your guess is right. The 100+ AS gain comes late so CQA has to decide before. I'll increase the timer to 60sec instead of 30, we'll see.
Memory : just tried, 4 days, 73MB. Something went wrong, maybe lots of actions on your side ? ARe your logs huge ?

**Comment by djozz911** [comment]
My logs at the time I wrote my comment :
![image](https://user-images.githubusercontent.com/47937938/67184057-27a8a400-f3e3-11e9-89b5-bfe3004409fd.png)

Now, another problem. At work, my machine is a Win7 pro and CQA works flawlessly . I'm on a Win10 pro at home and every time I use the auto event I get this error (found in ExceptionLog) and CQA closes : 

21/10/2019 09:05:22
System.ArgumentNullException: La valeur ne peut pas être null.
Nom du paramètre : source
   à System.Linq.Enumerable.Select[TSource,TResult](IEnumerable`1 source, Func`2 selector)
   à CQFollowerAutoclaimer.AutoEvent.<EventTimer_Elapsed>d__9.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_1(Object state)
   à System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(Object state)
   à System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   à System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   à System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   à System.Threading.ThreadPoolWorkQueue.Dispatch()
   à System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()

it happens right after the PG cards check (ActionLog) : 

21/10/2019 09:05:21 	 Sending CC catcher score 228
21/10/2019 09:05:22	Debug in PGCards, PGCards attempts remaining : no

Entries from Event Viewer : 
Application : CQ Auto-follower-claimer-inator.exe Version du Framework :  v4.0.30319 Description : le processus a été arrêté en raison d'une exception non  gérée. Informations sur l'exception : System.ArgumentNullException à  System.Linq.Enumerable.Select[[System.Int32, mscorlib, Version=4.0.0.0,  Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib,  Version=4.0.0.0, Culture=neutral,  PublicKeyToken=b77a5c561934e089]](System.Collections.Generic.IEnumerable`1<Int32>,  System.Func`2<Int32,System.__Canon>) à  CQFollowerAutoclaimer.AutoEvent+<EventTimer_Elapsed>d__9.MoveNext() à  System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_1(System.Object)  à System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(System.Object)  à  System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext,  System.Threading.ContextCallback, System.Object, Boolean) à  System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext,  System.Threading.ContextCallback, System.Object, Boolean) à  System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()  à System.Threading.ThreadPoolWorkQueue.Dispatch() à  System.Threading._ThreadPoolWaitCallback

and

Nom de l’application défaillante CQ Auto-follower-claimer-inator.exe, version : 4.3.3.0, horodatage : 0x5d972806
Nom du module défaillant : KERNELBASE.dll, version : 10.0.18362.418, horodatage : 0x2b181c2c
Code d’exception : 0xe0434352
Décalage d’erreur : 0x00113572
ID du processus défaillant : 0x390
Heure de début de l’application défaillante : 0x01d587896304c742
Chemin d’accès de l’application défaillante : C:\Users\Famille\Downloads\calc\CQ Auto-follower-claimer-inator.exe
Chemin d’accès du module défaillant: C:\Windows\System32\KERNELBASE.dll
ID de rapport : a1f91bbc-cb47-4d69-9731-34f6d6fe44e8
Nom complet du package défaillant : 
ID de l’application relative au package défaillant : 



I'll try on my other system (Win10 pro also) and edit this post.

**Comment by MatthieuBonne** [comment]
Are you sure you used latest version ?
A new one should be out in 3-4 days, I'll try to prevent these crashes.

**Comment by djozz911** [comment]
I'm using v4.5.0.0 from october 5.
Downloading latest version

edit : no crash with latest version. 'Désolé du dérangement' in french in the text ;o)
edit2 : note to self : always use the latest version before asking question !

**Comment by djozz911** [comment]
> Adventure : your guess is right. The 100+ AS gain comes late so CQA has to decide before. I'll increase the timer to 60sec instead of 30, we'll see.

Today, CQA shown "Adventure active, under progress" but didn't started. I had to do it manually.
Upon completion, the 60 sec timer didn't prevented CQA to start a PG adventure.

**Comment by MatthieuBonne** [comment]
Lots of fixes in 4.5.0.2, please try it. :)
